server/src/main.rs

```diff
@@ -45,6 +46,19 @@ use crate::storage::StorageEngine;
 #[global_allocator]
 static ALLOC: snmalloc_rs::SnMalloc = snmalloc_rs::SnMalloc;
 
+/// Folders for storing metadata and Apache Parquet files.
+pub struct DataFolders {
+    /// Folder for storing metadata and Apache Parquet files on the local file
+    /// system.
+    pub local_data_folder: PathBuf, // PathBuf to support complex operations.
```

Comment:
I had originally intended to use [`ObjectStore`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html) for all of the data folders so files would always be accessed through the same interface. However, even when using [`LocalFileSystem`](https://docs.rs/object_store/latest/object_store/local/struct.LocalFileSystem.html) directly it does not seem possible to extract the path set through [`LocalFileSystem::new_with_prefix()`](https://docs.rs/object_store/latest/object_store/local/struct.LocalFileSystem.html#method.new_with_prefix) for [`MetadataManager.metadata_database_path`](https://github.com/ModelarData/ModelarDB-RS/blob/master/server/src/metadata/mod.rs#L89) as [`LocalFileSystem.config`](https://docs.rs/object_store/latest/src/object_store/local.rs.html#191-193) is not public. We could technically extract the path through [`to_string()`](https://docs.rs/object_store/latest/src/object_store/local.rs.html#200-204) but we would have to remove the additional characters and changes to [`std::fmt::Display`](https://docs.rs/object_store/latest/src/object_store/local.rs.html#200-204) could easily cause problems that cannot be detected at compile time. So I do not think we can use [`ObjectStore`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html) exclusively as long as we use [`rusqlite`](https://docs.rs/rusqlite/latest/rusqlite/). Of course the [`StorageEngine`](https://github.com/ModelarData/ModelarDB-RS/blob/master/server/src/storage/mod.rs#L72) also currently requires that the local data folder is passed as a [`PathBuf`](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html), but hopefully this can be changed to an [`ObjectStore`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html) as we convert it to use asynchronous I/O instead of synchronous I/O. So in the future we may need to store `local_data_folder` as both a [`ObjectStore`](https://docs.rs/object_store/latest/object_store/trait.ObjectStore.html) and a [`PathBuf`](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html) in `DataFolders` to serve different components in the system.

Reply:
That makes sense.